src/overlay/Peer.cpp

```diff
@@ -852,6 +854,16 @@ Peer::recvError(StellarMessage const& msg)
 void
 Peer::noteHandshakeSuccessInPeerRecord()
 {
+    if (getIP().empty() || getRemoteListeningPort() == 0)
```

Comment:
I think we should allow 0 as a valid listening port. It just means that we can't contact them. For example if they are behind NAT.